enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/grpc/ExtAuthService.java

```diff
@@ -180,4 +194,34 @@ private int getCode(int statusCode) {
         }
         return Code.INTERNAL_VALUE;
     }
+
+    private String constructQueryParamString(String requestPath, Map<String, String> queryParamMap,
+                                             List<String> queryParamsToRemove) {
+        // If no query parameters needs to be removed, then the request path can be applied as it is.
+        if (queryParamsToRemove.size() == 0) {
+            return requestPath;
+        }
+
+        String pathWithoutQueryParams = requestPath.split("\\?")[0];
+        StringBuilder requestPathBuilder = new StringBuilder(pathWithoutQueryParams);
+        int count = 0;
+        if (queryParamMap.size() > 0) {
+            for (String queryParam : queryParamMap.keySet()) {
+                if (queryParamsToRemove.contains(queryParam)) {
+                    continue;
+                }
+                if (count == 0) {
+                    requestPathBuilder.append("?");
+                } else {
+                    requestPathBuilder.append("&");
+                }
+                requestPathBuilder.append(queryParam);
+                if (queryParamMap.get(queryParam) != null) {
```

Comment:
Is it okay for query value being an empty string? Just to clarify.

Reply:
https://www.rfc-editor.org/rfc/rfc3986#section-3.4 as per spec, it does not limit